api/graphql/types/recurring_reservation/serializers.py

```diff
@@ -437,3 +343,276 @@ def save(self, **kwargs: Any) -> RecurringReservation:
+            buffer_time_after=buffer_time_after,
+            skip_dates=skip_dates,
+            ignore_reservations=old_reservation_ids,
+        )
+
+        if slots.overlapping:
+            msg = "Not all reservations can be made due to overlapping reservations."
+            extensions = {
+                "code": error_codes.RESERVATION_SERIES_OVERLAPS,
+                "overlapping": slots.overlapping_json,
+            }
+            raise GraphQLError(msg, extensions=extensions)
+
+        # Remove CONFIRMED reservations that have not yet begun.
+        # Also remove any statistics for these reservations, since they are being replaced by the new ones.
+        old_reservations = instance.reservations.filter(begin__gt=now, state=ReservationStateChoice.CONFIRMED)
+        ReservationStatistic.objects.filter(reservation__in=old_reservations).delete()
+        old_reservations.delete()
+
+        return instance.actions.bulk_create_reservation_for_periods(slots.non_overlapping, reservation_details)
+
+    def get_reservation_details(self, instance: RecurringReservation) -> ReservationDetails:
+        """
+        Use reservation details from the next reservation relative to the current time that is going to occur.
+        If there is no next reservation, use the previous reservation.
+        If there are not reservations this way, check cancelled and denied reservations as well.
+
+        We validated previously that there is at least one reservation in the series,
+        so we can safely assume that we will have at least one reservation to get details from.
+        """
+        now = local_datetime()
+
+        next_reservation = instance.reservations.filter(begin__gte=now).going_to_occur().order_by("begin").first()
+        if next_reservation is None:
+            next_reservation = instance.reservations.filter(begin__lt=now).going_to_occur().order_by("-begin").first()
+        if next_reservation is None:
+            next_reservation = instance.reservations.filter(begin__gte=now).order_by("begin").first()
+        if next_reservation is None:
+            next_reservation = instance.reservations.filter(begin__lt=now).order_by("-begin").first()
+
+        return ReservationDetails(
+            name=next_reservation.name,
+            description=next_reservation.description,
+            num_persons=next_reservation.num_persons,
+            # If we needed to use a cancelled or denied reservation,
+            # state for the new reservation should be CONFIRMED instead.
+            state=(
+                next_reservation.state
+                if next_reservation.state in ReservationStateChoice.states_going_to_occur
+                else ReservationStateChoice.CONFIRMED
+            ),
+            type=next_reservation.type,
+            working_memo=next_reservation.working_memo,
+            #
+            handled_at=next_reservation.handled_at,
+            confirmed_at=next_reservation.confirmed_at,
```

Comment:
Both` handled_at` & `confirmed_at` can be set to current time to reflect when the series was rescheduled.

Reply:
Done.